pkg/cloud/scalewayprovider.go

```diff
@@ -0,0 +1,199 @@
+
+	"github.com/kubecost/opencost/pkg/log"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type ScalewayPricing struct {
+	NodesInfos map[string]*instance.ServerType
+	PVCost     float64
+}
+
+type Scaleway struct {
+	Clientset               clustercache.ClusterCache
+	Config                  *ProviderConfig
+	Pricing                 map[string]*ScalewayPricing
+	DownloadPricingDataLock sync.RWMutex
+}
+
+func (c *Scaleway) DownloadPricingData() error {
+	c.DownloadPricingDataLock.Lock()
+	defer c.DownloadPricingDataLock.Unlock()
+
+	// TODO wait for an official Pricing API from Scaleway
+	// Let's use a static map and an old API
+
+	if len(c.Pricing) != 0 {
+		// Already initialized
+		return nil
+	}
+
+	// PV pricing per AZ
+	pvPrice := map[string]float64{
+		"fr-par-1": 0.00011,
+		"fr-par-2": 0.00011,
+		"fr-par-3": 0.00032,
+		"nl-ams-1": 0.00008,
+		"nl-ams-2": 0.00008,
+		"pl-waw-1": 0.00011,
+	}
+
+	c.Pricing = make(map[string]*ScalewayPricing)
+
+	client, err := scw.NewClient(scw.WithAuth("SCWXXXXXXXXXXXXXXXXX", "00000000-0000-0000-0000-000000000000"), scw.WithDefaultProjectID("00000000-0000-0000-0000-000000000000"))
+	if err != nil {
+		return err
+	}
+
+	instanceAPI := instance.NewAPI(client)
+
+	for _, zone := range scw.AllZones {
+		resp, err := instanceAPI.ListServersTypes(&instance.ListServersTypesRequest{Zone: zone})
+		if err != nil {
+			log.Errorf("Could not get Scaleway pricing data from instance API in zone %s: %+v", zone, err)
+			continue
+		}
+		c.Pricing[zone.String()] = &ScalewayPricing{
+			PVCost:     pvPrice[zone.String()],
+			NodesInfos: map[string]*instance.ServerType{},
+		}
+
+		for name, infos := range resp.Servers {
+			c.Pricing[zone.String()].NodesInfos[name] = infos
+		}
+	}
+
+	return nil
+}
+
+type scalewayKey struct {
+	Labels map[string]string
+}
+
+func (k *scalewayKey) Features() string {
+	instanceType, _ := util.GetInstanceType(k.Labels)
+	zone, _ := util.GetZone(k.Labels)
+
+	return zone + "," + instanceType
+}
+
+func (k *scalewayKey) GPUType() string {
+	instanceType, _ := util.GetInstanceType(k.Labels)
+	if strings.HasPrefix(instanceType, "RENDER") || strings.HasPrefix(instanceType, "GPU") {
+		return instanceType
+	}
+	return ""
+}
+func (k *scalewayKey) ID() string {
+	return ""
+}
+
+func (c *Scaleway) NodePricing(key Key) (*Node, error) {
+	// There is only the zone and the instance ID in the providerID, hence we must use the features
+	split := strings.Split(key.Features(), ",")
+	if pricing, ok := c.Pricing[split[0]]; ok {
+		if info, ok := pricing.NodesInfos[split[1]]; ok {
+			log.Infof("Using features:`%s`", key.Features())
+			return &Node{
+				Cost:        fmt.Sprintf("%f", info.HourlyPrice),
+				PricingType: DefaultPrices,
+				VCPU:        fmt.Sprintf("%d", info.Ncpus),
+				RAM:         fmt.Sprintf("%d", info.RAM),
+				// This is tricky, as instances can have local volumes or not
```

Comment:
Makes sense; we later try and populate this via a Prometheus metric so don't feel wedded to this number!